src/youtube/js/youtube.js

```diff
@@ -0,0 +1,374 @@
+/*
+ * Youtube video plugin
+ */
+
+require("../../../html5-common/js/utils/InitModules/InitOO.js");
+require("../../../html5-common/js/utils/InitModules/InitOOUnderscore.js");
+require("../../../html5-common/js/utils/InitModules/InitOOHazmat.js");
+require("../../../html5-common/js/utils/constants.js");
+(function(_, $) {
+ 
+  var pluginName = "ooyalaYoutubeVideoTech";
+  var player;
+  var youtubePlayer;
+  var youtubeVideoContainer;
+  var element;
+  var youtubeID;
+  var playerNotready = true;
+  var playerReady = false;
+  var bitrateFlag = true;
+  var javascriptCommandQueue = [];
+  var qualities = [];
+
+  /**
+   * @class OoyalaYoutubeVideoFactory
+   * @classdesc Factory for creating video player objects for youtube videos.
+   * @property {string} name The name of the plugin
+   * @property {object} encodings An array of supported encoding types (ex. OO.VIDEO.ENCODING.MP4)
+   * @property {object} features An array of supported features (ex. OO.VIDEO.FEATURE.BITRATE_CONTROL)
+   * @property {string} technology The core video technology (ex. OO.VIDEO.TECHNOLOGY.HTML5)
+   */
+  var OoyalaYoutubeVideoFactory = function() {
+    this.name = pluginName;
+    this.features = [ OO.VIDEO.FEATURE.VIDEO_OBJECT_SHARING_GIVE,
+                      OO.VIDEO.FEATURE.BITRATE_CONTROL ];
+    this.technology = OO.VIDEO.TECHNOLOGY.HTML5;
+    this.encodings = [OO.VIDEO.ENCODING.MP4];
+
+    /**
+     * Creates a video player instance using OoyalaYoutubeVideoWrapper.
+     * @public
+     * @method OoyalaYoutubeVideoFactory#create
+     * @param {object} parentContainer The jquery div that should act as the parent for the video element
+     * @param {string} domId The id of the video player instance to create
+     * @param {object} controller A reference to the video controller in the Ooyala player
+     * @param {object} css The css to apply to the video element
+     * @param {string} playerId An id that represents the player instance
+     * @returns {object} A reference to the wrapper for the newly created element
+     */
+    this.create = function(parentContainer, domId, controller, css, playerId) {
+      youtubeVideoContainer = parentContainer;
+      player= document.createElement('div');
+      player.id ="player";
+      player.style.position = "absolute";
+      player.style.top = "0px";
+      player.style.left = "0px";
+
+      $('head').append("<script type = 'text/javascript' src = 'http://www.youtube.com/iframe_api'></script>");
+      window.onYouTubePlayerAPIReady = function() { onYouTubeIframeAPIReady(); }; 
```

Comment:
onYouTubePlayerAPIReady is a event published by youtube for creating the youtube player which needs to be listened using  window.onYouTubePlayerAPIReady. So we should not use the onYouTubePlayerAPIReady event name anywhere else for other events.